ENIGMAsystem/SHELL/Graphics_Systems/General/GSscreen.cpp

```diff
@@ -351,4 +372,104 @@ void screen_redraw()
   screen_refresh();
 }
 
+int screen_save(string filename) { //Assumes native integers are little endian
+  draw_batch_flush(batch_flush_deferred);
+
+  unsigned int fw = 0, fh = 0;
+  bool flipped = false;
+  unsigned char* rgba = enigma::graphics_copy_screen_pixels(&fw,&fh,&flipped);
+  int ret = image_save(filename, rgba, fw, fh, fw, fh, flipped);
+
+  delete[] rgba;
+  return ret;
+}
+
+int screen_save_part(string filename,unsigned x,unsigned y,unsigned w,unsigned h) { //Assumes native integers are little endian
+  draw_batch_flush(batch_flush_deferred);
+
+  bool flipped = false;
+  unsigned char* rgba = enigma::graphics_copy_screen_pixels(x,y,w,h,&flipped);
+  int ret = image_save(filename, rgba, w, h, w, h, flipped);
+
+  delete[] rgba;
+  return ret;
+}
+
+int background_create_from_screen(int x, int y, int w, int h, bool removeback, bool smooth, bool preload)
+{
+  draw_batch_flush(batch_flush_deferred);
+
+  bool flipped = false;
+  unsigned char* rgba = enigma::graphics_copy_screen_pixels(x,y,w,h,&flipped);
+
+  if (flipped)
+    rgba = enigma::image_flip(rgba, w, h, 4);
+
+  enigma::backgroundstructarray_reallocate();
+  int bckid=enigma::background_idmax;
+  enigma::background_new(bckid, w, h, &rgba[0], removeback, smooth, preload, false, 0, 0, 0, 0, 0, 0);
+  delete[] rgba;
+  enigma::background_idmax++;
+  return bckid;
+}
+
+int sprite_create_from_screen(int x, int y, int w, int h, bool removeback, bool smooth, bool preload, int xorig, int yorig) {
+  draw_batch_flush(batch_flush_deferred);
+
+  bool flipped = false;
+  unsigned char* rgba = enigma::graphics_copy_screen_pixels(x,y,w,h,&flipped);
+
+  if (flipped)
+    rgba = enigma::image_flip(rgba, w, h, 4);
+
+  enigma::spritestructarray_reallocate();
+  int sprid=enigma::sprite_idmax;
+  enigma::sprite_new_empty(sprid, 1, w, h, xorig, yorig, 0, h, 0, w, preload, smooth);
+  enigma::sprite_set_subimage(sprid, 0, w, h, rgba, rgba, enigma::ct_precise); //TODO: Support toggling of precise.
+  delete[] rgba;
+  return sprid;
+}
+
+int sprite_create_from_screen(int x, int y, int w, int h, bool removeback, bool smooth, int xorig, int yorig) {
+  return sprite_create_from_screen(x, y, w, h, removeback, smooth, true, xorig, yorig);
+}
+
+void sprite_add_from_screen(int id, int x, int y, int w, int h, bool removeback, bool smooth) {
+  draw_batch_flush(batch_flush_deferred);
+
+  bool flipped = false;
+  unsigned char* rgba = enigma::graphics_copy_screen_pixels(x,y,w,h,&flipped);
+
+  if (flipped)
+    rgba = enigma::image_flip(rgba, w, h, 4);
+
+  enigma::sprite_add_subimage(id, w, h, rgba, rgba, enigma::ct_precise); //TODO: Support toggling of precise.
+  delete[] rgba;
+}
+
+// The following two leave a bad taste in my mouth because they depend on views, which should be removable.
```

Comment:
You have solved this problem, and the comment is stale.

Reply:
Alright: b2a071293cbe7e9bf5b8d804b672a6cd0ebedb84

It's gone!